test/jdk/java/lang/Float/SixteenBitFormats.java

```diff
@@ -0,0 +1,263 @@
+                               " got 0x" + Integer.toHexString(actual));
+            return 1;
+            }
+        return 0;
+    }
+
+    private static int compareAndReportError(float input,
+                                             short expected) {
+        // Round to nearest even is sign symmetric
+        return compareAndReportError0( input, expected) +
+               compareAndReportError0(-input, Binary16.negate(expected));
+    }
+
+    private static int compareAndReportError0(short input,
+                                              float expected) {
+        float actual = Float.binary16AsShortBitsToFloat(input);
+        if (Float.compare(actual, expected) != 0) {
+            System.out.println("Unexpected result of converting " +
+                               Integer.toHexString(input & 0xFFFF) +
+                               " to float. Expected " + Float.toHexString(expected) +
+                               " got " + Float.toHexString(actual));
+            return 1;
+            }
+        return 0;
+    }
+
+    private static int compareAndReportError(short input,
+                                             float expected) {
+        // Round to nearest even is sign symmetric
+        return compareAndReportError0(                input,   expected) +
+               compareAndReportError0(Binary16.negate(input), -expected);
+    }
+
+    public static class Binary16 {
+        public static final short POSITIVE_INFINITY = (short)0x7c00;
+        public static final short NEGATIVE_INFINITY = (short)0xfc00;
+        public static final short MAX_VALUE = 0x7bff;
+        public static final short ONE = 0x3c00;
+        public static final short MIN_NORMAL = 0x0400;
+        public static final short MIN_VALUE = 0x0001;
+        public static final short NEGATIVE_ZERO = (short)0x8000;
+        public static final short POSITIVE_ZERO = 0x0000;
+        public static final short MAX_SUBNORMAL = (short)0x03ff;
+
+        public static boolean isNaN(short binary16) {
+            return ((binary16 & 0x7c00) == 0x7c00) // Max exponent and...
+                && ((binary16 & 0x03ff) != 0 );    // significand nonzero.
+        }
+
+        public static short negate(short binary16) {
+            return (short)(((binary16 & 0x8000) ^ 0x8000) | // Isolate and flip sign bit
+                           (binary16 & 0x7fff));
+        }
+
+        public static int compare(short bin16_1, short bin16_2) {
+            if (bin16_1 == bin16_2) {
+                return 0;
+            } else {
+                if (isNaN(bin16_1)) {
+                    return isNaN(bin16_2) ? 0 : 1;
+                } else {
+                    if (isNaN(bin16_2)) {
+                        return -1;
+                    }
+                    return Integer.compare((int)(bin16_1 & 0xFFFF),
+                                           (int)(bin16_2 & 0xFFFF));
+                }
+            }
+        }
+    }
+}
```

Comment:
The last invocation to `Integer.compare()` is not correct.
For example, if `bin16_1 = -1 `and `bin16_2 = 1`, the invocation returns 1, which is incorrect.